src/os/layer/preset/abstractpresetservice.js

```diff
@@ -0,0 +1,192 @@
+goog.module('os.layer.preset.AbstractPresetService');
+
+
+const IPresetService = goog.requireType('os.layer.preset.IPresetService');
+
+/**
+ * Class to provide stubs and simple implementation of setDefault() and setPublished()
+ *
+ * @abstract
+ * @implements {IPresetService}
+ */
+class AbstractPresetService {
+  /**
+   * @inheritDoc
+   */
+  insert(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  update(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  remove(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  find(search) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * The compiler is wrong; preset, opt_boolean, and the return are exactly as they should be
+   * @suppress {checkTypes}
```

Comment:
`checkTypes` suppressions are VERY strongly discouraged, to the point where we basically won't allow them anymore. Whatever compiler error is happening should be fixed correctly.

Reply:
Even with doing the JSDoc cast on every parameter and the return, GCC still thought that the params and return were "?".  So, instead of @inheritDoc, just added the same JSDoc as ipresetservice.js.